server/api_archive.go

```diff
@@ -0,0 +1,126 @@
+package server
+
+// This is a low-level archiver which zips and sends GCA server persistent file
+// data using a POST command. This is intended to be used by an external service
+// to provide archival snapshots of raw data in the system. Only public data is
+// provided, with private keys zeroed before sending.
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/glowlabs-org/gca-backend/glow"
+)
+
+var (
+	ApiArchiveRateLimiter *glow.RateLimiter // Provide a rate limiter for this API.
+)
+
+func init() {
+	ApiArchiveRateLimiter = glow.NewRateLimiter(apiArchiveLimit, apiArchiveRate)
+}
+
+// ArchiveHandler provides the POST /archive api endpoint. It returns
+// uninterpreted file data intended to be used by an external service
+// to produce an archive.
+func (gcas *GCAServer) ArchiveHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, "Only POST method is allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	if r.ContentLength != 0 {
+		http.Error(w, "Request should not have a body", http.StatusBadRequest)
+		return
+	}
+
+	if !ApiArchiveRateLimiter.Allow() {
+		http.Error(w, "Too many requests", http.StatusTooManyRequests)
+		return
+	}
+
+	buffer := new(bytes.Buffer)
+	zipw := zip.NewWriter(buffer)
+
+	files := []string{"server.keys", "gcaTempPubKey.dat", "gcaPubKey.dat", "equipment-authorizations.dat", "allDeviceStats.dat", "equipment-reports.dat"}
```

Comment:
This should probably be a const somewhere. Also need to update the readme to mention that this list needs to be updated every time that a new file is added to the GCA server.

I don't think you can make const slices, but you can make const arrays, and I'm pretty sure it should be fine to use an array here rather than a slice.